### ISSUE TITLE
Lack Of Mod Use Please Read Request ~ Integration Request

### ISSUE THREAD

**Issue by jredfox** [CAD97/SpawnerCraft#7]
Hi I am the creator of silk spawners. It's a bit more sophisticated then this and has some more features.  With silk spawners you can just pick up the actual spawners with silk touch.  I saw this mod and thought I would ask would you like to merge with my mod? Why because, my mod adds in dungeon tweeks that any living mob could be in a dungeon even modded unless configed otherwise.  Also it supports any mod that adds in there own spawner if it is defined as a spawner in my DefineSpawners.txt.  I saw this simple addition and would like to add this with a slightly different twist.  Rather then throwing a mob spawner on a stick at an entity you get a looting sword and then it will have a chance to drop 1/4 pieces for the beginning set.  With my mod you actually pick up and can place the actual mob spawner. My mod supports forge mobs does yours? Forge mobs are mobs without a global id. My mod also supports nei.  **If you integrate with my mod it will work with any mob regardless if they have an egg or not since my mod adds in forge eggs for forge mobs or forge eggs for entities that don't have an egg. It will try to drop your vanilla egg method else it will drop a forge egg or caged egg depending on if it's a living or non living entity. You can find the mod here**: http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/2743544-silk-spawners-not-enough-items-support . I have an idea of how your mod works. You get the egg from the vanilla egg (color first and second layer) then it either creates an item with new metadata or it just displays the egg based on nbt.  If you integrate with my mod It will do the same thing except if it can't find an egg it will be a forge egg part.

**Comment by jredfox** [comment]
well since your idea of mob spawn egg fractions came from the same plugin we both stole their ideas from a bucket plugin and it's not copyrighted I will just add in my spawn fraction egg system.

**Comment by CAD97** [comment]
@jredfox I claim no right to the idea of this mod; in fact, as this mod is [GPL](https://github.com/CAD97/SpawnerCraft/blob/1.10/LICENSE), so you are free to use code from this mod verbatim so long as you follow the [terms](https://www.gnu.org/licenses/quick-guide-gplv3.html) of said license.

If you provide a source repository link to the source code of your mod, I'd be happy to work with you to interface between the two mods; however, I'm very busy with college right now and without a clear path cannot afford time to working on potentially fruitless extensions to this mod. Keeping it working properly in the latest Forge and following good-neighbor practices is important, interfacing with another mod (that potentially serves as an alternative) is not.

Additionally, the point of this mod is to provide a minimal-extension way of utilizing spawners of mobs that have spawn eggs. As such this mod will likely never grow in scope. However, it should be pulling information from the Forge egg registry; all good-neighbor forge mobs with spawn eggs should work with this mod.

Separately, if you license your mod under (L)GPL and host the source on GitHub, I'd be open to helping you maintain it. But for the time being, I don't see any reason to work with your mod. The two mods have two different solutions to a similar problem, with slightly different ideals about how that solution should be structured. This mod will maintain its own identity, seperate from yours.

In any case, I'm closing this until at the very least you properly open-source (aka provide a public source repository link, not say "decompile") your mod, as until that point I can't do anything anyway. Sorry for the potentially repetitive wall of text this was written in one pass.

**Comment by jredfox** [comment]
You don't understand by integrate I mean literally take your mechanics from your egg system along with literally taking the idea of the mechanic system for spawners.  A full integration will provide publicity for my mod and a great deal of praise for your mod. **I would prefer my own eula which specifies it simply you may modify this mod and not claim it as your own, you may mirror the link, notch has permission over this mod, Microsoft may add this mod to vanilla.**.   _I don't really need that much help except an explination of how you got the drops dropping from the entities, I can then modify it appropriately and also how you got the textures_.

Here is my SRC:  https://github.com/jredfox/silkspawners  
My MC forum:  http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/2743544-silk-spawners-not-enough-items-support 
Eula: is on my mc forum in the spoiler called Eula

" To propagate" a work means to do anything with it that, without
permission, would make you directly or secondarily liable for
infringement under applicable copyright law, except executing it on a
computer or modifying a private copy.  Propagation includes copying,
distribution (with or without modification), making available to the
public, and in some countries other activities as well."

**Forge egg registry**. Well what if the mob doesn't have a spawn egg for the forge mob? My mod will support 1.9 format egg code soon enough once I get 1.7.10 fully updated and finished or at least the mounting update.  My mod has it's own egg system that isn't tied to metadata or hashmaps, It adds creative eggs to the creative tabs as new stacks with different nbt but, they themselves are not a hashmap and can be called anywhere for any nbt. My mod egg system will work weather or not a modder has placed an egg into the registry.

**The egg soul parts (4 to make one spawn egg) is your idea**.  I would prefer permision before copying the technical aspect of your mod to my mod. I would first play this mod see how it works and have and basically change it into my own system if the config for forging spawners is enabled. I also like the idea of mob spawn egg parts. I am curious as to how you are texturing the eggs is it by getting egg info from the hashmap?  Well It's not all your idea with both stole it from this one plugin idea http://imgur.com/a/AMxuk . Now I was unaware that there was a plugin called silk spawners untill a month after I named it. It didn't show up on google when I searched the name. It's also not a mod and the plugin doesn't work I hear.  So I am keeping my mod's name

**If we integrate into one mod this will be the single most popular spawner mod and dungeon tweaks mod on the forums**. I suggest we integrate. I will do 90% of the coding however with the publicity of us combined we can get me a beta tester so little or no bugs will appear as new features are implamented.  I don't catch every bug usually until days to weeks later due to lack of beta testers. Take a look at my configuration files in the spoiler.  There is a couple more config file systems that are not on the forum as I have not updated them this week. Note the newer features is 1.7.10 till I get it completly working. I would aprecieate some help in areas I am blank in like item interact event  setting the block and cancleing the event(NEI support one of the first bugs of the mod).  Now it does work with nei however it just doesn't work if the mob doesn't have a global id for 1.8+ since my work around was if it doesn't have a global id drop forge spawner but, it doesn't work with 1.8+ NEI.   My mod for 1.8+ fixes vanilla and people can place spawners in without cheat mod on. 

**"Separately, if you license your mod under (L)GPL and host the source on GitHub, I'd be open to helping you maintain it. But for the time being, I don't see any reason to work with your mod. The two mods have two different solutions to a similar problem, with slightly different ideals about how that solution should be structured. This mod will maintain its own identity, separate from yours."**. 

I suggest an integration with forging spawners enabled by default. Looting enchantment to get spawn egg parts unless configed to your mob spawner on a fishing pole.  Is that the only Issue I am a modder that ads in a config option for any technical aspect of a technical mod.  

In conclusion let's integrate.  I only have one term of condition we keep a simpler less strict Eula, and keep the name **Silk Spawners**.   Do you except? Well if not will you at least let me adapt the part egg system and is it ok for people to forge their own vanilla blank mob spawners. Forge spawners is my idea but, the vanilla forging was your idea.

**Comment by CAD97** [comment]
OK, I think I understand your request properly now.

**I (@CAD97) hereby grant you (@jredfox) the permission to use any ideas and / or code from my mod SpawnerCraft, under the following conditions:**
**1. The source code of any derivative work is available in a public repository**
**2. There is no fee charged to use the derivative work**

However, I will continue to maintain this mod separately, as I enjoy doing so and I desire the smaller scope. In any case, I doubt "integrating" would benefit either of us any, as though I have [40,000 downloads](https://minecraft.curseforge.com/projects/spawnercraft/files?filter-game-version=&filter-status=), no more than 8,000 come from a single release and I have no community built around the mod. And, if integration is to provide publicity to _your_ mod, how will it bring "a great deal of praise to [my] mod"?

So in conclusion: no, I will not discontinue my mod to integrate with yours, but I give you permission to use concepts from this mod to enhance yours.

And as an addendum, I would not roll your own license. A specific Open Source license is much more legally secure. You can still add an addendum to the license that Notch and/or Mojang may take full ownership of the work (though, technically, Mojang may already be able to do whatever they want with direct-Minecraft-derivatives).

---

For adding drops to Entities you don't control, see [DropsListener::onMobDrops](https://github.com/CAD97/SpawnerCraft/blob/1.10/src/main/java/cad97/spawnercraft/handler/DropsListener.java#L29). For render colors, see [SpawnerCraftItem::registerColors](https://github.com/CAD97/SpawnerCraft/blob/1.10/src/main/java/cad97/spawnercraft/init/SpawnerCraftItems.java#L54).

**Comment by jredfox** [comment]
Question by derivative.  The source of my mod is on github but, isn't the vanilla egg parts system that exact same way? Do I have permission to use the base model for your egg and some of the texture code. The mod is free of charge? no mod can be charged to you mean no adfly links?  

By the way if you would have integrated your mod wouldn't have died and you could also do some of the work but, I understand. You want people to have options.  

 I am saying people can miror this so if it disappears  on the forum someone else can go to a mirror and download it. The source code is in the mod also.  

I saw the two added links I will check them out.  again do I have permission for the base model of the egg I really like it and I can't do a better job even if I tried. 

I will always have an open source for silk spawners as it is so complex the only way it's getting updated is that if someone has the original source. It's on gibhub always and it's always in the .jar file itself.  I mostly use github for an issue tracker.

**O and can you say themodpack maker since people don't know me as jredfox on here that's my mc forum name.**

**Comment by CAD97** [comment]
**For the purposes of the above granting of permission, GitHub user @jredfox and MinecraftForum user @themodpackmaker are the same person. Additionally, where the original says "ideas and / or code", use instead "ideas and / or code and / or other assets which belong to the author".**

Derivative = Using content that comes from the prior work.  
Free of charge = no monetary barrier to use (affiliate links, adfly, etc are allowed).

Note that the code compiled in the .jar is very different from the code that you wrote; information is lost and obscured by the compilation step.

My egg model was generated by re-texturing quartz and ripping the spawn_egg texture directly from the game's assets.

My intent for this mod was never for it to be popular; it was made by me for my own benefit, and to provide an option for augmenting the base game without changing it too much. In any case, I'm done with this thread, as all that needs to be said has been.
